libs/libc/sched/task_atexit.c

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ * libs/libc/sched/task_atexit.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdlib.h>
+#include <assert.h>
+#include <unistd.h>
+#include <debug.h>
+#include <errno.h>
+
+#include <nuttx/sched.h>
+
+#ifdef CONFIG_SCHED_ATEXIT
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nxtask_atexit_proxy
+ *
+ * Description:
+ *    A wrapper function to call atexit callback in onexit.
+ *
+ * Input Parameters:
+ *   exitcode - The exit status of task
+ *   arg      - The data pointer from user space
+ *
+ ****************************************************************************/
+
+static void nxtask_atexit_proxy(int exitcode, FAR void *arg)
+{
+  (*(atexitfunc_t)arg)();
+}
```

Comment:
Hmmm... I don't think this will work.  Will it?

This will lie in the kernel address space, but will be called in user mode.  I would think that would cause a crash.

Am I missing something.